app/routes/application_status.py

```diff
@@ -1,34 +1,29 @@
 from flask_restful import Resource
+from flask import current_app
 from zeep.exceptions import Fault
 
 from ..libs import VivaApplicationStatus
-from ..libs import hash_to_personal_number
+from ..libs import decode_hash_id
+from ..errors import ValidationError
 
 
 class ApplicationStatus(Resource):
 
     def get(self, hash_id):
-        try:
-            if not hash_id:
-                raise TypeError('hash_id should be type string')
+        hash_id_tuple = decode_hash_id(hash_id=hash_id)
 
-            personal_number = hash_to_personal_number(hash_id=hash_id)
+        if len(hash_id_tuple) == 0:
+            raise ValidationError(
+                message=f'The hashid {hash_id} in the url is not a valid hashid')
```

Comment:
happy hacker :)

Reply:
Agree with this! No need to print any unsanitized input 👍 .